### PR TITLE
SHANK-71 $export kickoff endpoint

### PR DESCRIPTION
**Pull request by dlennon-va**
Some new application properties - that someone will need to add to the deployment:

identityservice.url=unset
identityservice.encodingKey=bulk-smash
identityservice.patientIdPattern=[0-9]+(V[0-9]{6})?

incrediblebulk.public-url=unset
incrediblebulk.public-bulk-status-path=/services/fhir/v0/dstu2/bulk

identityservice.url, identityservice.patientIdPattern, incrediblebulk.public-bulk-status-path are all defaulted in the base application.properties so may not need to be configured. encodingKey is as well, but if we want it different per environment we should configure it. incredblebulk.public-url is the only thing for sure we will need.

### CODE REVIEW

File: the-incredible-bulk/src/main/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientController.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  private final PublicationStatusTransformer transformer;
+
+  private final String bulkStatusUrl;
+
+  private final IdentityService identityService;
+
+  @Builder
+  BulkPatientController(
+      @Value("${incrediblebulk.url}") String baseUrl,
+      @Value("${incrediblebulk.bulk-status-path:/services/fhir/v0/dstu2/bulk}")
```
Comment (ilaflamme-va) [inline the-incredible-bulk/src/main/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientController.java]: This path doesn't seem right. The _/services_  part of our paths' is only external, i.e. through adhoc gateway. It is stripped off way before it reaches the applications.

File: the-incredible-bulk/src/main/resources/application.properties [hunk 2]
```diff
@@ -36,6 +44,8 @@ spring.jpa.show-sql=true
 logging.level.liquibase.executor=INFO
 
 bulk.file.writer=local
+incrediblebulk.url=unset
+incrediblebulk.bulk-status-path=/services/fhir/v0/dstu2/bulk
```
Comment (ilaflamme-va) [inline the-incredible-bulk/src/main/resources/application.properties]: No _/services_

File: the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                    .buildStartEpoch(now)
+                    .recordsPerFile(10)
+                    .build()));
+    when(request.getRequestURI()).thenReturn("/Patient/$export");
+    when(request.getQueryString()).thenReturn("_outputFormat=ndjson");
+    when(identityService.register(anyList()))
+        .thenReturn(List.of(Registration.builder().uuid("ZZZ").build()));
+    Map<String, String> headers = new HashMap<>();
+    headers.put("accept", "application/fhir+json");
+    headers.put("prefer", "respond-async");
```
Comment (ilaflamme-va) [inline the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java]: A helper method might remove some of the duplication of data initialization for these tests.

File: the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    assertThat(response.getStatusCodeValue()).isEqualTo(503);
+    assertThat(response.getBody()).isNotNull();
+    assertThat(response.getBody().text()).isNotNull();
+    assertThat(response.getBody().text().div()).contains("try again later");
+  }
+
+  @Test
+  void exportWithInvalidAcceptHeaderReturnsBadRequest() {
+    Map<String, String> headers = new HashMap<>();
+    headers.put("accept", "y u no like me");
```
Comment (bschofield-va) [inline the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java]: ... sad ...

File: bulk-fhir-ids-mapping/pom.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <artifactId>api-starter</artifactId>
+    <version>3.1.12</version>
+    <relativePath/>
+  </parent>
+  <artifactId>bulk-fhir-ids-mapping</artifactId>
+  <version>1.0.18-SNAPSHOT</version>
+  <packaging>jar</packaging>
+  <properties>
+    <health-apis-ids.version>2.1.3</health-apis-ids.version>
+    <jacoco.coverage>0.94</jacoco.coverage>
```
[the remark this answers is not in the document]
Comment (dlennon-va) [reply bulk-fhir-ids-mapping/pom.xml]: It shall be .95!

File: the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java (hunk 3 above)
Comment (dlennon-va) [reply the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java]: Yep, good call, moved the common header setup to another method.

**Pull request merged**